lib/factory_bot_rails/railtie.rb

```diff
@@ -21,7 +22,13 @@ class Railtie < Rails::Railtie
     end
 
     config.after_initialize do |app|
-      FactoryBot.find_definitions
+      if app.config.factory_bot.lazy_load_definitions && !app.config.eager_load
+        require "factory_bot_rails/lazy_registry_find"
+        FactoryBot::Registry.prepend FactoryBotRails::LazyRegistryFind
```

Comment:
This should probably be implemented in the `factory_bot` gem instead, so I'll open another PR for that if there's any interest here.